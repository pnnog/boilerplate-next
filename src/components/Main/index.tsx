type MainProps = {
  title: string;
};

export default function Main({ title }: MainProps) {
  return (
    <div>
      <h1>{title}</h1>
    </div>
  );
}
