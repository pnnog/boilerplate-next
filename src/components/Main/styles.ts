import styled from 'styled-components';

export const Wrapper = styled.div`
  background-color: green;
`;
export const Title = styled.h1`
  color: red;
`;
