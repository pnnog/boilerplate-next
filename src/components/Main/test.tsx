import { render, screen } from '@testing-library/react';
import Main from '.';

const props = {
  title: 'Main Title'
};

describe('Main', () => {
  it('should render main title', () => {
    render(<Main {...props} />);
    expect(
      screen.getByRole('heading', {
        name: /Main Title/i
      })
    ).toBeInTheDocument();
  });

  it('should render title with blue color', () => {
    const { container } = render(<Main {...props} />);

    expect(screen.getByRole('heading', { name: props.title })).toHaveStyle({
      color: 'red'
    });

    expect(container.firstElementChild).toMatchSnapshot();
  });
});
