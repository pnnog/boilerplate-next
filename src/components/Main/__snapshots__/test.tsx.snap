// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render title with blue color 1`] = `
.c0 {
  background-color: green;
}

.c1 {
  color: red;
}

<div
  class="c0"
>
  <h1
    class="c1"
  >
    Main Title
  </h1>
</div>
`;
